File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.py
class Solution(object):
    def checkDivisibility(self, n):
        org=n
        dsum=0
        pro=1
        while n!=0:
            digit=n%10
            dsum+=digit
            pro*=digit
            n//=10
        return org%(dsum+pro)==0   
        

File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.java
class Solution {
    public boolean checkDivisibility(int n) {
        int org=n;
        int dsum=0;
        int pro=1;
        while(n!=0){
            int digit=n%10;
            dsum+=digit;
            pro*=digit;
            n/=10;
        }
        return org%(dsum+pro)==0;
        
    }
}